File: services/location-service/controllers/locationController.js
const { Storage } = require('@google-cloud/storage');
const admin = require('firebase-admin');
require('dotenv').config();
const { v4: uuidv4 } = require("uuid");
const {getCoordinateFromAdress} = require("../../../helpers/getCoordinatesFromAdress")

// Initialize Firestore and Storage
const db = admin.firestore();
const storage = new Storage();

exports.checkLocation = async (req,res)=> {
    const { input } = req.query;

  if (!input) {
    return res.status(400).json({ error: 'Input is required' });
  }

  try {
    const response = await fetch(
      `https://nominatim.openstreetmap.org/search?format=json&q=${input}&limit=1`
    );
    const data = await response.json();
    res.json({ isValid: data.length > 0 });
  } catch (error) {
    console.error('Error validating address:', error);
    res.status(500).json({ error: 'Internal server error' });
  }
}

// Assuming you are using Express.js
exports.giveCoordinates = async (req, res) => {
  const { address } = req.body;

  if (!address) {
    return res.status(400).json({ error: 'Address is required' });
  }

  try {
    const coordinates = await getCoordinateFromAdress(address);
    console.log(coordinates)
    
    // Check if coordinates were returned successfully
    if (!coordinates) {
      return res.status(404).json({ error: 'Coordinates not found for the given address' });
    }

    return res.status(200).json({ coordinates });
  } catch (error) {
    console.error('Error fetching coordinates:', error);
    return res.status(500).json({ error: 'An error occurred while fetching coordinates' });
  }
};

File: services/location-service/controllers/locationController.py
from flask import Flask, request, jsonify
import requests
import os
from dotenv import load_dotenv
import uuid
import firebase_admin
from firebase_admin import firestore
from google.cloud import storage
from helpers.getCoordinatesFromAdress import get_coordinate_from_address

# Load environment variables from a .env file
load_dotenv()

# Initialize Firestore and Storage
firebase_admin.initialize_app()
db = firestore.client()
storage_client = storage.Client()

app = Flask(__name__)

@app.route('/checkLocation', methods=['GET'])
def check_location():
    input_address = request.args.get('input')

    if not input_address:
        return jsonify({'error': 'Input is required'}), 400

    try:
        response = requests.get(
            f'https://nominatim.openstreetmap.org/search?format=json&q={input_address}&limit=1'
        )
        data = response.json()
        return jsonify({'isValid': len(data) > 0})
    except Exception as error:
        print('Error validating address:', error)
        return jsonify({'error': 'Internal server error'}), 500

@app.route('/giveCoordinates', methods=['POST'])
def give_coordinates():
    address = request.json.get('address')

    if not address:
        return jsonify({'error': 'Address is required'}), 400

    try:
        coordinates = get_coordinate_from_address(address)
        print(coordinates)

        if not coordinates:
            return jsonify({'error': 'Coordinates not found for the given address'}), 404

        return jsonify({'coordinates': coordinates}), 200
    except Exception as error:
        print('Error fetching coordinates:', error)
        return jsonify({'error': 'An error occurred while fetching coordinates'}), 500

if __name__ == '__main__':
    app.run(debug=True)
